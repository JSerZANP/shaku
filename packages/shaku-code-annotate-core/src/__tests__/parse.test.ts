import { test, describe, expect } from "vitest";
import { parseLine, shouldApplyAnnotation } from "../parser";

describe("parseLine() can parse comment lines", () => {
  const inputs = [
    "  -------  ",
    "  -------!",
    "  ------- !",
    "  -------<  ",
    "  -------<! ",
    "  ------- <<  ",
    "  ------- <<  ",
    "  ------- << !  ",
    "  ~~~~~~~~<  ",
    "  ~~~~~~~~ << ",
    "  ......  ",
    "  ......! ",
    "  ......!! ",
    "  ......<  ",
    "  ......<!  ",
    "  ...... <<  ",
    "  ...... <<!  ",
    "   [This is annotation]  ",
    "   [This is annotation]!  ",
    "   [This is annotation]<  ",
    "   [This is annotation]< !",
    "   [This is annotation] <<<  ",
    "   [This is annotation] <<< !  ",
    "   ^  ",
    "   ^!  ",
    "   ^ !  ",
    "   ^<  ",
    "   ^<!  ",
    "   ^<!!  ",
    "   ^ <<< ",
    "   ^ <<<! ",
    "   ^ <<< ! ",
    "   @fold start ",
    "   @fold start! ",
    "   @fold start ! ",
    "   @fold end ",
    "   @fold end !",
    "   @fold unexpected ",
    "   @fold unexpected !",
    "   @highlight start ",
    "   @highlight start! ",
    "   @highlight end ",
    "   @highlight end ! ",
    "   @highlight ",
    "   @highlight !",
    "   @highlight below ",
    "   @highlight below !",
    "   @highlight unexpected ",
    "   @highlight unexpected !",
    "   @dim start ",
    "   @dim start !",
    "   @dim end ",
    "   @dim end !",
    "   @dim ",
    "   @dim !",
    "   @dim below",
    "   @dim below !",
    "   @focus start ",
    "   @focus start!",
    "   @focus end ",
    "   @focus end ! ",
    "   @focus ",
    "   @focus !",
    "   @focus below ",
    "   @focus below! ",
    "   @focus below ",
    "()",
    "() !",
    "()()()",
    "()()() !",
    "()()() !\n",
    " ( )  ( )   ()   ",
    " ( )  ( )   () !  ",
    "(1)",
    "(1)!",
    "(1)(1)(10)",
    "(1)(1)(10)\n",
    "(1)(1)(10) !",
    " (0)  (2 )   (0)   ",
    " (0)  (2 )   (0)   !",
    " ((",
    " ((!",
    " (0)  (2 )   0",
    " (0)  (2 )   0!",
    " (0)  (2 )<",
    " (0)  (2 )<!",
    "   (0)  (2 ) <<<",
    "   (0)  (2 ) <<<  !",
    "@diff ",
    " @diff !",
    "   @diff + ",
    "   @diff  -  ",
    "   @diff - !",
    "   @diff + below",
    "   @diff + start",
    "   @diff  -  end   ",
    "   @diff unexpected ",
    "@diff unexpected !",
    " @diff + unexpected ",
    "    @highlight v   ",
    "    @highlight ^",
    " @dim v",
    " @dim ^ ",
    "   @focus v  ",
    " @focus ^",
    " @diff   +   v   ",
    "   @diff   +   ^   ",
    "@diff - v",
    "@diff -   ^  ",
    "   (r)(g)(b) ",
    " (red) (green) (blue) ",
    "@class a",
    "@class a!  ",
    "@class abc efg h123   ",
    "@class abc efg h123!",
    "@class abc-1 efg-2 h123!",
  ];
  for (const input of inputs) {
    test(input, () => {
      expect(parseLine(input)).toMatchSnapshot();
    });
  }

  test("   @dim unexpected ", () => {
    expect(parseLine("   @dim unexpected ")).toBeNull();
  });

  test("   @focus unexpected ", () => {
    expect(parseLine("   @focus unexpected ")).toBeNull();
  });

  test("This is some comments from source code ", () => {
    const result = parseLine("This is some comments from source code");
    expect(result).toBeNull();
  });
});

describe("shouldApplyAnnotation() should work", () => {
  test(" annotate ", () => {
    expect(shouldApplyAnnotation(" annotate ")).toBeTruthy();
  });

  test(" ", () => {
    expect(shouldApplyAnnotation("  ")).toBeFalsy();
  });
});
