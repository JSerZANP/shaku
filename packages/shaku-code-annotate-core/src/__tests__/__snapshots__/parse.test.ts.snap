// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseLine() can parse comment lines >     @highlight ^ 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >     @highlight v    1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    (0)  (2 ) <<<  ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 5,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >    (0)  (2 ) <<< 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 5,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >    (r)(g)(b)  1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "r",
        "length": 3,
        "offset": 3,
      },
      {
        "id": "g",
        "length": 3,
        "offset": 6,
      },
      {
        "id": "b",
        "length": 3,
        "offset": 9,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >    @collapse end  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
    "offset": 3,
  },
  "type": "DirectiveCollapse",
}
`;

exports[`parseLine() can parse comment lines >    @collapse end ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "end",
    "offset": 3,
  },
  "type": "DirectiveCollapse",
}
`;

exports[`parseLine() can parse comment lines >    @collapse start  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
    "offset": 3,
  },
  "type": "DirectiveCollapse",
}
`;

exports[`parseLine() can parse comment lines >    @collapse start !  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "start",
    "offset": 3,
  },
  "type": "DirectiveCollapse",
}
`;

exports[`parseLine() can parse comment lines >    @collapse start!  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "start",
    "offset": 3,
  },
  "type": "DirectiveCollapse",
}
`;

exports[`parseLine() can parse comment lines >    @collapse unexpected  1`] = `null`;

exports[`parseLine() can parse comment lines >    @collapse unexpected ! 1`] = `null`;

exports[`parseLine() can parse comment lines >    @diff   +   ^    1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
    "type": "+",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff  -   1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
    "type": "-",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff  -  end    1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
    "type": "-",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff +  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
    "type": "+",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff + below 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
    "type": "+",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff + start 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
    "type": "+",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff - ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
    "type": "-",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >    @diff unexpected  1`] = `null`;

exports[`parseLine() can parse comment lines >    @dim  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim below ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim below 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim end  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim end ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "end",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim start  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @dim start ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "start",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >    @focus  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus below  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus below  2`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus below!  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus end  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus end !  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "end",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus start  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus start! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "start",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @focus v   1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines >    @highlight  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight below  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "below",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight below ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "below",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight end  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight end !  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "end",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight start  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight start!  1`] = `
{
  "config": {
    "isEscaped": true,
    "mark": "start",
  },
  "type": "DirectiveHighlight",
}
`;

exports[`parseLine() can parse comment lines >    @highlight unexpected  1`] = `null`;

exports[`parseLine() can parse comment lines >    @highlight unexpected ! 1`] = `null`;

exports[`parseLine() can parse comment lines >    [This is annotation]   1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": false,
    "offset": 3,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    [This is annotation] <<<   1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": false,
    "offset": 0,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    [This is annotation] <<< !   1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": true,
    "offset": 0,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    [This is annotation]!   1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": true,
    "offset": 3,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    [This is annotation]<   1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": false,
    "offset": 2,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    [This is annotation]< ! 1`] = `
{
  "config": {
    "content": "This is annotation",
    "isEscaped": true,
    "offset": 2,
  },
  "type": "AnnotationLine",
}
`;

exports[`parseLine() can parse comment lines >    ^   1`] = `
{
  "config": {
    "isEscaped": false,
    "offset": 3,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^ !   1`] = `
{
  "config": {
    "isEscaped": true,
    "offset": 3,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^ <<<  1`] = `
{
  "config": {
    "isEscaped": false,
    "offset": 0,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^ <<< !  1`] = `
{
  "config": {
    "isEscaped": true,
    "offset": 0,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^ <<<!  1`] = `
{
  "config": {
    "isEscaped": true,
    "offset": 0,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^!   1`] = `
{
  "config": {
    "isEscaped": true,
    "offset": 3,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^<   1`] = `
{
  "config": {
    "isEscaped": false,
    "offset": 2,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^<!   1`] = `
{
  "config": {
    "isEscaped": true,
    "offset": 2,
  },
  "type": "DirectiveCallout",
}
`;

exports[`parseLine() can parse comment lines >    ^<!!   1`] = `null`;

exports[`parseLine() can parse comment lines >   ......   1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": false,
    "offset": 2,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ...... <<   1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": false,
    "offset": 0,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ...... <<!   1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": true,
    "offset": 0,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ......!  1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": true,
    "offset": 2,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ......!!  1`] = `null`;

exports[`parseLine() can parse comment lines >   ......<   1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": false,
    "offset": 1,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ......<!   1`] = `
{
  "config": {
    "content": "......",
    "isEscaped": true,
    "offset": 1,
    "style": "dotted",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ~~~~~~~~ <<  1`] = `
{
  "config": {
    "content": "~~~~~~~~",
    "isEscaped": false,
    "offset": 0,
    "style": "wavy",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ~~~~~~~~<   1`] = `
{
  "config": {
    "content": "~~~~~~~~",
    "isEscaped": false,
    "offset": 1,
    "style": "wavy",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   -------   1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": false,
    "offset": 2,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ------- ! 1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": true,
    "offset": 2,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ------- <<   1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": false,
    "offset": 0,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ------- <<   2`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": false,
    "offset": 0,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   ------- << !   1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": true,
    "offset": 0,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   -------! 1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": true,
    "offset": 2,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   -------<   1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": false,
    "offset": 1,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >   -------<!  1`] = `
{
  "config": {
    "content": "-------",
    "isEscaped": true,
    "offset": 1,
    "style": "solid",
  },
  "type": "DirectiveUnderline",
}
`;

exports[`parseLine() can parse comment lines >  ( )  ( )   ()    1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": undefined,
        "length": 3,
        "offset": 1,
      },
      {
        "id": undefined,
        "length": 3,
        "offset": 6,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 12,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  ( )  ( )   () !   1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": undefined,
        "length": 3,
        "offset": 1,
      },
      {
        "id": undefined,
        "length": 3,
        "offset": 6,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 12,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  (( 1`] = `null`;

exports[`parseLine() can parse comment lines >  ((! 1`] = `null`;

exports[`parseLine() can parse comment lines >  (0)  (2 )   (0)    1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 1,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 6,
      },
      {
        "id": "0",
        "length": 3,
        "offset": 13,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  (0)  (2 )   (0)   ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 1,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 6,
      },
      {
        "id": "0",
        "length": 3,
        "offset": 13,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  (0)  (2 )   0 1`] = `null`;

exports[`parseLine() can parse comment lines >  (0)  (2 )   0! 1`] = `null`;

exports[`parseLine() can parse comment lines >  (0)  (2 )< 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 5,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  (0)  (2 )<! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": "0",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "2",
        "length": 4,
        "offset": 5,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  (red) (green) (blue)  1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "red",
        "length": 5,
        "offset": 1,
      },
      {
        "id": "green",
        "length": 7,
        "offset": 7,
      },
      {
        "id": "blue",
        "length": 6,
        "offset": 15,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines >  @diff   +   v    1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
    "type": "+",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines >  @diff ! 1`] = `null`;

exports[`parseLine() can parse comment lines >  @diff + unexpected  1`] = `null`;

exports[`parseLine() can parse comment lines >  @dim ^  1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >  @dim v 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
  },
  "type": "DirectiveDim",
}
`;

exports[`parseLine() can parse comment lines >  @focus ^ 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
  },
  "type": "DirectiveFocus",
}
`;

exports[`parseLine() can parse comment lines > () ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": undefined,
        "length": 2,
        "offset": 0,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > () 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": undefined,
        "length": 2,
        "offset": 0,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > ()()() !
 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": undefined,
        "length": 2,
        "offset": 0,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 2,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 4,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > ()()() ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": undefined,
        "length": 2,
        "offset": 0,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 2,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 4,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > ()()() 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": undefined,
        "length": 2,
        "offset": 0,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 2,
      },
      {
        "id": undefined,
        "length": 2,
        "offset": 4,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > (1) 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "1",
        "length": 3,
        "offset": 0,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > (1)! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": "1",
        "length": 3,
        "offset": 0,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > (1)(1)(10)
 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "1",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "1",
        "length": 3,
        "offset": 3,
      },
      {
        "id": "10",
        "length": 4,
        "offset": 6,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > (1)(1)(10) ! 1`] = `
{
  "config": {
    "isEscaped": true,
    "parts": [
      {
        "id": "1",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "1",
        "length": 3,
        "offset": 3,
      },
      {
        "id": "10",
        "length": 4,
        "offset": 6,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > (1)(1)(10) 1`] = `
{
  "config": {
    "isEscaped": false,
    "parts": [
      {
        "id": "1",
        "length": 3,
        "offset": 0,
      },
      {
        "id": "1",
        "length": 3,
        "offset": 3,
      },
      {
        "id": "10",
        "length": 4,
        "offset": 6,
      },
    ],
  },
  "type": "DirectiveHighlightInline",
}
`;

exports[`parseLine() can parse comment lines > @diff  1`] = `null`;

exports[`parseLine() can parse comment lines > @diff -   ^   1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "end",
    "type": "-",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines > @diff - v 1`] = `
{
  "config": {
    "isEscaped": false,
    "mark": "start",
    "type": "-",
  },
  "type": "DirectiveDiff",
}
`;

exports[`parseLine() can parse comment lines > @diff unexpected ! 1`] = `null`;
